src/mpi/init/init_topo.c

```diff
@@ -29,50 +29,10 @@
 
 void MPII_init_topo(void)
 {
-#ifdef HAVE_HWLOC
-    MPIR_Process.bindset = hwloc_bitmap_alloc();
-    hwloc_topology_init(&MPIR_Process.hwloc_topology);
-    MPIR_Process.bindset_is_valid = 0;
-    hwloc_topology_set_io_types_filter(MPIR_Process.hwloc_topology, HWLOC_TYPE_FILTER_KEEP_ALL);
-    if (!hwloc_topology_load(MPIR_Process.hwloc_topology)) {
-        MPIR_Process.bindset_is_valid =
-            !hwloc_get_proc_cpubind(MPIR_Process.hwloc_topology, getpid(), MPIR_Process.bindset,
-                                    HWLOC_CPUBIND_PROCESS);
-    }
-#endif
-
-#ifdef HAVE_NETLOC
-    MPIR_Process.network_attr.u.tree.node_levels = NULL;
-    MPIR_Process.network_attr.network_endpoint = NULL;
-    MPIR_Process.netloc_topology = NULL;
-    MPIR_Process.network_attr.type = MPIR_NETLOC_NETWORK_TYPE__INVALID;
-    if (strlen(MPIR_CVAR_NETLOC_NODE_FILE)) {
-        mpi_errno =
-            netloc_parse_topology(&MPIR_Process.netloc_topology, MPIR_CVAR_NETLOC_NODE_FILE);
-        if (mpi_errno == NETLOC_SUCCESS) {
-            MPIR_Netloc_parse_topology(MPIR_Process.netloc_topology, &MPIR_Process.network_attr);
-        }
-    }
-#endif
+    MPIR_Hardware_topology_init();
```

Comment:
You may as well remove `init_topo.c` and directly call `MPIR_Hardware_topology_...()` directly. `init_topo.c` was just a placeholder when I did the refactor before a more thorough investigation.

Reply:
Right, I forgot about that one, I will add another commit for that